internal/servers/repository_workerauth.go

```diff
@@ -0,0 +1,843 @@
+package servers
+
+import (
+	"context"
+	"crypto/x509"
+	"database/sql"
+	"fmt"
+	"strconv"
+
+	"github.com/fatih/structs"
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/servers/store"
+	"github.com/hashicorp/boundary/internal/types/scope"
+	"github.com/hashicorp/go-dbw"
+	wrapping "github.com/hashicorp/go-kms-wrapping/v2"
+	nodee "github.com/hashicorp/nodeenrollment"
+	"github.com/hashicorp/nodeenrollment/types"
+	"github.com/mitchellh/mapstructure"
+	"google.golang.org/protobuf/proto"
+)
+
+// Ensure we implement the Storage interfaces
+var (
+	_ nodee.Storage = (*WorkerAuthRepositoryStorage)(nil)
+)
+
+type certAuthVersion struct {
+	Version uint32
+}
+
+// WorkerAuthRepositoryStorage is the Worker Auth database repository
+type WorkerAuthRepositoryStorage struct {
+	reader db.Reader
+	writer db.Writer
+	kms    *kms.Kms
+}
+
+// NewRepositoryStorage creates a new WorkerAuthRepositoryStorage that implements the Storage interface
+func NewRepositoryStorage(ctx context.Context, r db.Reader, w db.Writer, kms *kms.Kms) (*WorkerAuthRepositoryStorage, error) {
+	const op = "workerauth.NewRepositoryStorage"
+	switch {
+	case isNil(r):
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "reader is nil")
+	case isNil(w):
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "writer is nil")
+	case isNil(kms):
+		return nil, errors.New(ctx, errors.InvalidParameter, op, "kms is nil")
+	}
+
+	workerAuthRepoStorage := &WorkerAuthRepositoryStorage{
+		reader: r,
+		writer: w,
+		kms:    kms,
+	}
+
+	return workerAuthRepoStorage, nil
+}
+
+// Store implements the Storage interface
+func (r *WorkerAuthRepositoryStorage) Store(ctx context.Context, msg nodee.MessageWithId) error {
+	const op = "workerauth.(WorkerAuthRepositoryStorage).Store"
+	if err := types.ValidateMessage(msg); err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	if msg.GetId() == "" {
+		return errors.New(ctx, errors.InvalidParameter, op, "given message cannot be stored as it has no ID")
+	}
+
+	// Determine type of message to store
+	marshaledBytes, err := proto.Marshal(msg.(proto.Message))
+	if err != nil {
+		return errors.New(ctx, errors.Encode, op, "error marshaling nodee.MessageWithId", errors.WithWrap(err))
+	}
+	switch msg.(type) {
+	case *types.NodeInformation:
+		node, err := unmarshalNodeInformation(ctx, marshaledBytes)
```

Comment:
How come you're marshaling and unmarshaling instead of just using the value directly?